### ISSUE TITLE
dataChecking function is ignore the data without filed.

### ISSUE THREAD

**Issue by ZoneTwelve** [Seal-Nexus/csv-database#3]
https://github.com/Seal-Nexus/csv-database/blob/aa627edba0d280e4de1a08503ad9a1040939ddaa/index.js#L248-L263

Normally data checking is required to check whether it's matching the rules or not, but with the rules (filed) missing, and data is not empty, the rule checking just returns the value back.
So, it requires to check the rules is existing or not is very important, with the database is not checking the rule, will cause another problem to happen.
Please remember to check the rule exists then check the data, and **not just returns the value back**.

following the example code can find this problem:
```javascript
#!/usr/bin/env node
const path = require("path");
const database = require("../index");
let db = new database({
  filePath: "$pid/$date",
  fileName: "database.csv",
  standDir: path.resolve( __dirname, "database" ),
  fileds: [
    { name: "pid",  type: "string" },
    { name: "hr",   type: "string" },
    { name: "spo2", type: "string" },
    { name: "date", type: "string" }, // user input date
    { name: "sys",  type: "string", autofill: ( ) => `${Number(new Date())}` },
  ]
});
async function main( ){
  let pid  = "p001",
      hr   = `${~~(Math.random()*30) + 70}`,
      spo2 =  `${~~(Math.random()*5) + 95}`,
      date = "2022-01-01";
  
  let write_res = await db.write({ pid, hr, spo2, date, misspell:date });
  console.log( { pid, hr, spo2, date } );
  console.log( "[Write]", write_res );

  let search_res = await db.read({ pid, date });
  console.log( "[Read]", search_res.length );
  
}
main( );
```

**Comment by ZoneTwelve** [comment]
Bug fix at [28e5dc7](https://github.com/Seal-Nexus/csv-database/commit/28e5dc7ba63d396e69aaf7b5e0747cf44c5a83b9)

**Comment by ZoneTwelve** [comment]
Bug fix at [28e5dc7](https://github.com/Seal-Nexus/csv-database/commit/28e5dc7ba63d396e69aaf7b5e0747cf44c5a83b9)
